src/core/qgsdiagramrenderer.h

```diff
@@ -546,26 +568,73 @@ class CORE_EXPORT QgsDiagramRenderer
      */
     void setSizeLegend( bool enabled ) { mShowSizeLegend = enabled; }
 
-    /** Returns the marker symbol used for rendering the diagram size legend.
+    /**
+     * Returns the marker symbol used for rendering the diagram size legend.
      * \since QGIS 2.16
      * \see setSizeLegendSymbol()
      * \see sizeLegend()
      */
     QgsMarkerSymbol *sizeLegendSymbol() const { return mSizeLegendSymbol.get(); }
 
-    /** Sets the marker symbol used for rendering the diagram size legend.
+    /**
+     * Sets the marker symbol used for rendering the diagram size legend.
      * \param symbol marker symbol, ownership is transferred to the renderer.
      * \since QGIS 2.16
      * \see sizeLegendSymbol()
      * \see setSizeLegend()
      */
-    void setSizeLegendSymbol( QgsMarkerSymbol *symbol ) { mSizeLegendSymbol.reset( symbol ); }
+    virtual void setSizeLegendSymbol( QgsMarkerSymbol *symbol ) { mSizeLegendSymbol.reset( symbol ); }
+
+    /**
+     * Set the label used as title or the size attribute
+     * @brief setSizeLabel
+     * @note added in QGIS 3.0
+     * @param label
+     */
+    void setSizeLabel( QString label ) { mSizeLabel = label; }
+
+    /**
+     * @brief sizeLabel
+     * @note added in QGIS 3.0
+     * @return
+     */
+    QString sizeLabel( ) const { return mSizeLabel; }
+
+    /**
+     * The size class used for the legend (label, value)
+     * @brief setSizeRules
+     * @note added in QGIS 3.0
+     * @param rules
+     */
+    void setSizeRules( QList< double > rules ) { mSizeRules = rules; }
+
+    /**
+     * @brief sizeRules
+     * @note added in QGIS 3.0
+     * @return
+     */
+    QList< double > sizeRules() const { return mSizeRules; }
+
+    /**
+     * @brief setSizeLegendType
+     * @note added in QGIS 3.0
+     * @param type
```

Comment:
Can you review the doxygen comments in this class? They don't need to be very long, but they should provide a bit more information than the function signature. It's also ok to explain the parameters directly in the text instead of a separate `\param` definition.

If you formulate them as

> The size legend \a type controls ... 

you can copy paste between getters and setters.

Also needs transforming of `@note added in` to `\since QGIS 3.0`